File: src/TestMagicSquares.java

public class TestMagicSquares {
	public static void main(String[] args) {
		MagicSquare square = MagicSquaresGenerator.generateSquare(7);
		System.out.println("The original square:");
		System.out.println(square);
		square.toPuzzle(5);
		System.out.println("The original square with hidden elements:");
		System.out.println(square.puzzleString());
		MagicSquare removed = square.copy();
		removed.removeInvisible(); 
		System.out.println("The original square with removed elements:");
		System.out.println(removed);
		//MagicSquareState start = removed.toMagicSquareState();
		//start.testSquare();
		//MagicSquareSolver solver = new MagicSquareSolver();
		//MagicSquareState finished = solver.solve(start);
		//System.out.println("The solved square:");
		//System.out.println(finished);
		//System.out.println("The squares are equal: "+square.equals(finished));
	}
}

File: cplusplussrc/TestMagicSquares.cpp
#include "TestMagicSquares.h"
#include "MagicSquaresGenerator.h"
#include "MagicSquareState.h"
#include "GeneralSolver.h"

void TestMagicSquares::main(std::string args[])
{
	MagicSquareState *square = MagicSquareState::toMagicSquareState(MagicSquaresGenerator::generateSquare(10));
	std::cout << std::string("The original square:") << std::endl;
	std::cout << square << std::endl;
	square->toPuzzle(1);
	std::cout << std::string("The original square with hidden elements:") << std::endl;
	std::cout << square->puzzleString() << std::endl;
	MagicSquareState *removed = square->copy();
	removed->removeInvisible();
	std::cout << std::string("The original square with removed elements:") << std::endl;
	std::cout << removed << std::endl;
	MagicSquareState *start = MagicSquareState::toMagicSquareState(removed);
	GeneralSolver *solver = new GeneralSolver();
	MagicSquareState *finished = solver->checkPossibilities(start);
	std::cout << std::string("The solved square:") << std::endl;
	std::cout << finished << std::endl;
	std::cout << std::string("The squares are equal: ") << square->equals(finished) << std::endl;
}
